### PR TITLE
EOS-27889 - Generate Ceph RPM’s from cortx-rgw repo

### PR DESCRIPTION
**Pull request by shailesh-vaidya**
# Problem Statement
- Build Ceph packages from cortx-rgw repository. 
- Generate cortx-rgw image on Rocky Linux 8.4
- Custom CI for Rocky Linux 8.4

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability -
        - cortx-rgw pachage and image genration - http://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/cortx-rgw-build/23/console
        - custom-ci on Rocky Linux 8.4  http://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/52/
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/custom-ci/generic/custom-ci.groovy [hunk 1]
```diff
@@ -37,8 +37,8 @@ pipeline {
         string(name: 'MOTR_URL', defaultValue: 'https://github.com/Seagate/cortx-motr.git', description: 'Motr Repository URL', trim: true)
         string(name: 'PRVSNR_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for Provisioner', trim: true)
         string(name: 'PRVSNR_URL', defaultValue: 'https://github.com/Seagate/cortx-prvsnr.git', description: 'Provisioner Repository URL', trim: true)
-        string(name: 'S3_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for S3Server', trim: true)
-        string(name: 'S3_URL', defaultValue: 'https://github.com/Seagate/cortx-s3server.git', description: 'S3Server Repository URL', trim: true)
+        string(name: 'CORTX_RGW_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for S3Server', trim: true)
+        string(name: 'CORTX_RGW_URL', defaultValue: 'https://github.com/Seagate/cortx-rgw', description: 'S3Server Repository URL', trim: true)
```
Comment (gauravchaudhari02) [inline jenkins/custom-ci/generic/custom-ci.groovy:41]: Description need to be changed for both parameters

File: jenkins/custom-ci/generic/motr.groovy [hunk 2]
```diff
@@ -16,8 +16,8 @@ pipeline {
     parameters {  
         string(name: 'MOTR_URL', defaultValue: 'https://github.com/Seagate/cortx-motr', description: 'Branch for Motr.')
         string(name: 'MOTR_BRANCH', defaultValue: 'main', description: 'Branch for Motr.')
-        string(name: 'S3_URL', defaultValue: 'https://github.com/Seagate/cortx-s3server', description: 'Branch for S3Server')
-        string(name: 'S3_BRANCH', defaultValue: 'main', description: 'Branch for S3Server')
+        string(name: 'CORTX_RGW_URL', defaultValue: 'https://github.com/Seagate/cortx-rgw', description: 'Branch for S3Server')
+        string(name: 'CORTX_RGW_BRANCH', defaultValue: 'main', description: 'Branch for S3Server')
```
Comment (gauravchaudhari02) [inline jenkins/custom-ci/generic/motr.groovy:20]: Description need to be changed for both parameters

**Review by AbhijitPatil1992 — APPROVED** [review]
Changes looks good

File: docker/cortx-deploy/build.sh [hunk 3]
```diff
@@ -54,10 +55,24 @@ if [ -z "${BUILD}" ] ; then
...
+fi
+
+if [ "$SERVICE" == "cortx-rgw" ] && [ "$OS" == "centos-7.9.2009" ]; then
+echo -e "#####################################################################"
+echo -e "# SERVICE: $SERVICE Image Build is NOT supported on $OS              "
+echo -e "#####################################################################"
+exit 1
+fi
+
+if [ "$SERVICE" == "cortx-rgw" ] && [ "$OS" == "centos-7.9.2009" ]; then
```
Comment (gauravchaudhari02) [inline docker/cortx-deploy/build.sh:71]: Why to same check condition twice?

File: docker/cortx-deploy/generate-release-info.sh [hunk 4]
```diff
@@ -30,7 +30,7 @@ VERSION: "$VERSION"
 OS: $(cat /etc/redhat-release | sed -e 's/ $//g' -e 's/^/\"/g' -e 's/$/\"/g')
 DATETIME: $(date +"%d-%b-%Y %H:%M %Z" | sed -e 's/^/\"/g' -e 's/$/\"/g')
 COMPONENTS:
-$(for component in $(sed 's/#.*//g' /opt/seagate/cortx/cortx-componenet-rpms.txt); do echo "    - \"$(rpm -q $component).rpm\"" ; done)
+$(for component in $(sed 's/#.*//g' /opt/seagate/cortx/cortx-componenet-rpms.txt ceph-componenet-rpms.txt); do echo "    - \"$(rpm -q $component).rpm\"" ; done)
```
Comment (gauravchaudhari02) [inline docker/cortx-deploy/generate-release-info.sh:33]: do we need to give full path for `ceph-componenet-rpms.txt`?

File: jenkins/custom-ci/generic/custom-ci.groovy (hunk 1 above)
Comment (shailesh-vaidya) [reply jenkins/custom-ci/generic/custom-ci.groovy]: Fixed

File: jenkins/custom-ci/generic/motr.groovy (hunk 2 above)
Comment (shailesh-vaidya) [reply jenkins/custom-ci/generic/motr.groovy]: Fixed

File: docker/cortx-deploy/build.sh (hunk 3 above)
Comment (shailesh-vaidya) [reply docker/cortx-deploy/build.sh]: Fixed

File: docker/cortx-deploy/generate-release-info.sh (hunk 4 above)
Comment (shailesh-vaidya) [reply docker/cortx-deploy/generate-release-info.sh]: Fixed

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**